net/interfaces/interfaces_linux.go

```diff
@@ -170,7 +172,25 @@ func defaultRouteInterfaceProcNet() (string, error) {
 	}
 
 	return "", errors.New("no default routes found")
+}
 
+func defaultRouteInterfaceProcNet() (string, error) {
+	rc, err := defaultRouteInterfaceProcNetInternal(128)
+	if rc == "" {
```

Comment:
rc == "" && err == nil ?

Reply:
Done. err == io.EOF when problem occurs, so check that err is either nil or io.EOF.